### PR TITLE
Add definition of RTCEncodedVideoFrameMetadata frameId and dependencies

### PR DESCRIPTION
**Pull request by tonyherre**
Fixes Issue #220

***
<a href="https://pr-preview.s3.amazonaws.com/tonyherre/webrtc-encoded-transform/pull/222.html" title="Last updated on Jan 23, 2024, 10:04 PM UTC (9691a88)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-encoded-transform/222/09f618e...tonyherre:9691a88.html" title="Last updated on Jan 23, 2024, 10:04 PM UTC (9691a88)">Diff</a>

### CODE REVIEW

**Comment by tonyherre** [comment]
@aboba, @alvestrand, @fippo PTAL

File: index.bs [hunk 1]
```diff
@@ -346,6 +346,25 @@ dictionary RTCEncodedVideoFrameMetadata {
 ### Members ### {#RTCEncodedVideoFrameMetadata-members}
 
 <dl dfn-for="RTCEncodedVideoFrameMetadata" class="dictionary-members">
+    <dt>
+        <dfn dict-member>frameId</dfn> <span class="idlMemberType">unsigned long long</span>
+    </dt>
+    <dd>
+        <p>
+            An identifier for the encoded frame, monotonically increasing in decode order. Its lower
+            16 bits matches the frame_number of the Dependency Descriptor RTP Header Extension, if sent.
```
[the remark this answers is not in the document]
Comment (fippo) [reply index.bs:355]: Not sure whether it should be "if received"? On the sender this can be present unconditionally?

File: index.bs [hunk 2]
```diff
@@ -346,6 +346,25 @@ dictionary RTCEncodedVideoFrameMetadata {
...
+            Only present for received frames if the Dependency Descriptor Header Extension is present.
+        </p>
+    </dd>
+    <dt>
+        <dfn dict-member>dependencies</dfn> <span class="idlMemberType">sequence&lt;unsigned long long&gt;</span>
+    </dt>
+    <dd>
+        <p>
+            List of frameIds of frames this frame references.
+            Only present for received frames if the Dependency Descriptor Header Extension is present.
```
Comment (fippo) [inline index.bs:365]:
```suggestion
            Only present for received frames if the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[?AV1-RTP]]  is present.
```

Comment (fippo) [reply index.bs:365]: (copied from webrtc-svc, I hope this resolves correctly)

**Review by fippo — APPROVED** [review]
LGTM with a question and suggestions

File: index.bs (hunk 2 above)
Comment (tonyherre) [reply index.bs]: Awesome thanks. I'd been wondering how to best reference these. Done.

File: index.bs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tonyherre) [reply index.bs]: Hmm, good point, this line about matching the frame_number should be true on both sender and receiver. Changed to just "if present", matching the next line.

File: index.bs [hunk 3]
```diff
@@ -346,6 +346,25 @@ dictionary RTCEncodedVideoFrameMetadata {
 ### Members ### {#RTCEncodedVideoFrameMetadata-members}
 
 <dl dfn-for="RTCEncodedVideoFrameMetadata" class="dictionary-members">
+    <dt>
+        <dfn dict-member>frameId</dfn> <span class="idlMemberType">unsigned long long</span>
+    </dt>
+    <dd>
+        <p>
+            An identifier for the encoded frame, monotonically increasing in decode order. Its lower
```
[the remark this answers is not in the document]
Comment (tonyherre) [reply index.bs:354]: The ID is still in decode order, whatever the receive order happens to be, as it's set by the encoder and propagated in the Dependency Descriptor.

File: index.bs (hunk 2 above)
Comment (fippo) [reply index.bs]: Doesn't render :-(
We need some magic similar to this:
https://github.com/w3c/webrtc-svc/blob/main/svc-respec-config.js#L43
which needs to be included like https://github.com/w3c/webrtc-svc/blob/main/index.html#L8C3-L8C63

Comment (fippo) [reply index.bs]: But this is bikeshed... try this https://speced.github.io/bikeshed/#biblio-custom ?

**Review by aboba — APPROVED** [review]
LGTM.

File: index.bs [hunk 4]
```diff
@@ -346,6 +346,25 @@ dictionary RTCEncodedVideoFrameMetadata {
...
+            16 bits match the frame_number of the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[?AV1-RTP]], if present.
```
Comment (fippo) [inline index.bs:355]:
```suggestion
            16 bits match the frame_number of the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[?AV1-RTP-SPEC]], if present.
```

File: index.bs [hunk 5]
```diff
@@ -346,6 +346,25 @@ dictionary RTCEncodedVideoFrameMetadata {
...
+            Only present for received frames if the Dependency Descriptor Header Extension is present.
+        </p>
+    </dd>
+    <dt>
+        <dfn dict-member>dependencies</dfn> <span class="idlMemberType">sequence&lt;unsigned long long&gt;</span>
+    </dt>
+    <dd>
+        <p>
+            List of frameIds of frames this frame references.
+            Only present for received frames if the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[?AV1-RTP]] is present.
```
Comment (fippo) [inline index.bs:365]:
```suggestion
            Only present for received frames if the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[?AV1-RTP-SPEC]] is present.
```

**Review by fippo — COMMENTED** [review]
Lets try AV1-RTP-SPEC which seems to be the canonical thing: https://www.specref.org/?q=av1

File: index.bs [hunk 6]
```diff
@@ -346,6 +346,25 @@ dictionary RTCEncodedVideoFrameMetadata {
 ### Members ### {#RTCEncodedVideoFrameMetadata-members}
 
 <dl dfn-for="RTCEncodedVideoFrameMetadata" class="dictionary-members">
+    <dt>
+        <dfn dict-member>frameId</dfn> <span class="idlMemberType">unsigned long long</span>
+    </dt>
+    <dd>
+        <p>
+            An identifier for the encoded frame, monotonically increasing in decode order. Its lower
+            16 bits match the frame_number of the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[?AV1-RTP-SPEC]], if present.
```
Comment (fippo) [inline index.bs:355]:
```suggestion
            16 bits match the frame_number of the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[AV1-RTP-SPEC]], if present.
```

File: index.bs [hunk 7]
```diff
@@ -346,6 +346,25 @@ dictionary RTCEncodedVideoFrameMetadata {
...
+            Only present for received frames if the Dependency Descriptor Header Extension is present.
+        </p>
+    </dd>
+    <dt>
+        <dfn dict-member>dependencies</dfn> <span class="idlMemberType">sequence&lt;unsigned long long&gt;</span>
+    </dt>
+    <dd>
+        <p>
+            List of frameIds of frames this frame references.
+            Only present for received frames if the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[?AV1-RTP-SPEC]] is present.
```
Comment (fippo) [inline index.bs:365]:
```suggestion
            Only present for received frames if the AV1 Dependency Descriptor Header Extension defined in Appendix A of [[AV1-RTP-SPEC]] is present.
```

**Review by fippo — COMMENTED** [review]
Less `?`

File: index.bs (hunk 6 above)
Comment (alvestrand) [reply index.bs:355]: W3C doesn't like normative dependencies on specs that are IPR-targets. If we can get away with this being an informative dependency, we may prefer that.

**Review by henbos — APPROVED** [review]

**Pull request merged**